### DISCUSSION TITLE
Malayalam Language Leader

### DISCUSSION THREAD

**Discussion by arunshekher** [OHF-Voice/intents#457] [Language Leaders]
I would like to apply and volunteer myself to Malayalam language leader role. I am currently contributing to Home Assistant frontend, Android and iOS companion apps' Malayalam translations.

**Comment by synesthesiam** [comment]
Thanks @arunshekher! I'll add you as language leader for Malayalam :+1:

**Reply by arunshekher** [reply]
Thank you @synesthesiam, May I know what does this role entail other than contributing to HA voice intents in Malayalam language with merge requests to the repository.

**Reply by arunshekher** [reply]
Got myself answered to a good extent from the recent GitHub notification emails and by going through other discussions in this forum.

**Reply by synesthesiam** [reply]
Great! As you probably figured out, this entails reviewing the PRs and merging them if they look good.

**Comment by arunshekher** [comment]
@synesthesiam Is it okay to add `todo:` comments in the `yaml` files to hint to potential future contributors as to what the current contributor is trying to achieve so that it encourages discussion beforehand?

**Reply by synesthesiam** [reply]
Let's add any TODOs in comments for now :+1:

**Reply by arunshekher** [reply]
You mean GitHub comments?

**Reply by synesthesiam** [reply]
In the YAML

**Reply by arunshekher** [reply]
Okay, got it. 👍🏽

**Comment by arunshekher** [comment]
@synesthesiam Now that the sentence validator works with _Indic_ scripts and all intents are almost completed for Malayalam, is it advised to translate the `expansion_rules` list `keys` as well ?

**Reply by synesthesiam** [reply]
Yes, please! All except `name` and `area`.

**Reply by arunshekher** [reply]
Okay, Thanks.

**Comment by arunshekher** [comment]
@synesthesiam, 

I see these strings: `&id001`  `*id001` that got added at line 11 and line 18 in the file `homeassistant_HassTurnOff.yaml` for **Malayalam** after PR #698 merge. 

Are they suppose to be there?



https://github.com/home-assistant/intents/blob/342fc33e88a7a2829325631055c493f89576f033/sentences/ml/homeassistant_HassTurnOff.yaml#L11-L19


Searching for it in this repo shows that these are present in [hr] and [hu] languages as well.

**Reply by synesthesiam** [reply]
No, thank you for catching this. That is just `pyyaml` trying to be more clever than it is :(

**Reply by synesthesiam** [reply]
They should just be deleted.

**Comment by arunshekher** [comment]
@synesthesiam Is there a strategy in the pipeline to programatically handle the presence/absence of a space character in a `device` or `entity` name to get the device/entity identified correctly? 

Now when the `device` name in interpreted and parsed without space character by the speech to text engine the intent does not work for **Malayalam** language.

![ezgif com-gif-maker (5)](https://user-images.githubusercontent.com/315195/215489066-46f4a49c-a31c-4a0f-a7fa-774566760c9d.png)

In this example I've created a `device` with the name `പരീക്ഷണ വിളക്ക്`

The intents works when the STT interprets the device name as `പരീക്ഷണ വിളക്ക്` but not when interpreted as `പരീക്ഷണവിളക്ക്` (without space in between the words)

I have to pause in between the utterance to get it interpreted with space.

Here is the screencast video with my voice commands:


https://user-images.githubusercontent.com/315195/215490904-77aa567c-96c8-4e90-9078-d52b7cb48b4e.mov

**Reply by synesthesiam** [reply]
There isn't a programmatic way to handle this yet, but it is something we will need to add. Are you able to add an alias without the space?

**Reply by arunshekher** [reply]
Thank you for clarification. I'm not sure if I know how to add an alias for a device yet. I will search HA documentation and forum and figure this out. 👍🏽

**Reply by synesthesiam** [reply]
In the web UI, click a device and then go to Settings, Advanced settings, Aliases

**Reply by arunshekher** [reply]
Thank you @synesthesiam, I'll test out with aliases. Also I happen to watch '[Year of the Voice - Chapter 1](https://www.youtube.com/watch?v=ixgNT3RETPg&t=9s)' YT video today and saw the use of aliases in that and learned lot more about the project.

**Comment by synesthesiam** [comment]
@arunshekher when you get a chance, could you please look into translating the `HassGetState` sentences/responses?

**Reply by arunshekher** [reply]
@synesthesiam sure, will do it before the end of the month.
